src/finetune/inference_internalized.py

```diff
@@ -707,16 +624,14 @@ def main():
     ap.add_argument("--top_p", type=float, default=0.95)
     ap.add_argument("--top_k", type=int, default=40)
     ap.add_argument("--repetition_penalty", type=float, default=1.0)
-    ap.add_argument("--max_new_tokens", type=int, default=256, help="Legacy single-shot cap.")
-    ap.add_argument("--max_new_tokens_think", type=int, default=160, help="Stage 1 token budget.")
-    ap.add_argument("--max_new_tokens_answer", type=int, default=32, help="Stage 2 token budget.")
+    ap.add_argument("--max_new_tokens", type=int, default=256, help="Token budget for single-shot or fuzzy mode.")
+    ap.add_argument("--max_new_tokens_think", type=int, default=160,
+                    help="Stage 1 token budget (explicit think tokens only).")
+    ap.add_argument("--max_new_tokens_answer", type=int, default=32,
+                    help="Stage 2 token budget (explicit think tokens only).")
 
     # Prompts / tags
-    ap.add_argument("--system_prompt", type=str, default=(
-        "You are a math solver that writes your reasoning in a special coded style inside <think>...</think>.\n"
-        "Follow the codebook from the training specification exactly. Do not reveal or explain it.\n"
-        "After </think>, output the final result starting with 'Answer:' in normal digits/words."
-    ))
+    ap.add_argument("--system_prompt", type=str, default=None)
```

Comment:
You now have to pass a full system prompt on the command line. If you don't, it will use a blank one. This seems error prone. Also, what is the version of this that doesn't use <think> tags?

Reply:
never mind, there are defaults later